modules/nixos.nix

```diff
@@ -0,0 +1,41 @@
+floxOverlay: nixpkgs:
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}:
+
+let
+  cfg = config.programs.flox;
+  # "Extract" _only_ flox package from floxOverlay.
+  flox = (import nixpkgs { inherit (pkgs) system; overlays = [floxOverlay]; }).flox;
+in
+{
+
+  options = {
+    programs.flox = {
+      enable = lib.mkEnableOption "Flox CLI - Harness the power of Nix";
+      package = lib.mkOption {
+        type = lib.types.package;
+        description = "Flox CLI package";
+        default = flox;
+        defaultText = lib.literalExpression "pkgs.flox";
+        example = lib.literalExpression "pkgs.flox";
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment.systemPackages = [ cfg.package ];
+    nix.settings = {
+      trusted-public-keys = lib.mkAfter [
+        "flox-cache-public-1:7F4OyH7ZCnFhcze3fJdfyXYLQw/aV7GEed86nQ7IsOs="
+      ];
+      substituters = lib.mkAfter [
+        "https://cache.flox.dev"
+      ];
+    };
+
+  };
+}
```

Comment:
style: since nixos.nix and darwin.nix are identical, it would be nice to deduplicate those files unless you're expecting them to change. Or to factor stuff out with home.nix as well since it's almost identical

Reply:
Yeah, I wasn't sure they would end up being identical but they :)

One other feature I plan to bring in is the possibility to run Flox containers as a systemd/launchd unit. That would look different between `nixos.nix` and `darwin.nix`.

Would it be OK to leave it like this for now?

Reply:
Yep!

Reply:
Yep!